src/DependencyInjection/Compiler/CommandHandlerPass.php

```diff
@@ -36,9 +41,25 @@ public function process(ContainerBuilder $container)
             $builder->registerInContainer($container, $routing->commandToServiceMapping($builder->id()));
         }
 
+        $this->setDebugReport($container, $builders, $routing);
+
         // Setup default aliases
         $container->setAlias('tactician.commandbus', $builders->defaultBus()->serviceId());
         $container->setAlias('tactician.handler.locator.symfony', $builders->defaultBus()->locatorServiceId());
         $container->setAlias('tactician.middleware.command_handler', $builders->defaultBus()->commandHandlerMiddlewareId());
     }
+
+    private function setDebugReport(ContainerBuilder $container, BusBuilders $builders, Routing $routing)
+    {
+        $report = new RoutingDebugReport($builders, $routing);
+        $container->set('tactician.debug.report', $report);
```

Comment:
This creates a synthetic service, that doesn't work at compile time (instances cannot be dumped, synthetics are just removed before dumping the container).
I think creating a parameter from all `BusBuilder` mappings is the way to go (bus builders should probably create/update this parameter internally in `BusBuilder::registerInContainer()`). 
Then you would just have to create the command service as you started below, but passing the parameter as constructor arg instead.